pages/curriculum/[lesson].tsx

```diff
@@ -6,33 +6,51 @@ import LessonTitleCard from '../../components/LessonTitleCard'
 import AlertsDisplay from '../../components/AlertsDisplay'
 import ChallengeMaterial from '../../components/ChallengeMaterial'
 import GET_APP from '../../graphql/queries/getApp'
-import { Lesson, LessonStatus } from '../../@types/lesson'
-import { UserSubmission } from '../../@types/challenge'
-import { AppData } from '../../@types/app'
+import {
+  Challenge,
+  Lesson,
+  UserLesson,
+  Submission,
+  GetAppQuery
+} from '../../graphql/index'
 import withQueryLoader, {
   QueryDataProps
 } from '../../containers/withQueryLoader'
 import _ from 'lodash'
 
-const Challenges: React.FC<QueryDataProps<AppData>> = ({ queryData }) => {
+const Challenges: React.FC<QueryDataProps<GetAppQuery>> = ({ queryData }) => {
   const { lessons, session, alerts } = queryData
   const router = useRouter()
-  const currentlessonId = router.query.lesson as string
+  const currentlessonId = Number(router.query.lesson) as number
   if (!lessons || !alerts) {
     return <Error code={StatusCode.INTERNAL_SERVER_ERROR} message="Bad data" />
   }
   const currentLesson: Lesson | undefined = lessons.find(
-    (lesson: Lesson) => lesson.id === currentlessonId
-  )
+    lesson => lesson.id === currentlessonId
+  ) as Lesson
```

Comment:
If you remove the type assertions this works without issues, the compiler can infer the type automatically.

```ts
const currentLesson = lessons.find(lesson => lesson.id === currentlessonId)
```